js/cms.js

```diff
@@ -54,6 +54,17 @@ var CMS = {
   posts: [],
   pages: [],
   loaded: {},
+  
+  models: {
+    file: {
+      name: '',
+      link: '',
+      date: undefined,
+      get url() {
+        return CMS.settings.mode == 'Github' ? this.link : this.name;
```

Comment:
I've encapsulated the logic about which url to use for a file into the `file` object, by moving it from the `getContent` function. Now, if you want to use the url for a `file` in another function, you don't have to duplicate the logic.